lib/ash/type/map.ex

```diff
@@ -1,13 +1,51 @@
 defmodule Ash.Type.Map do
+  @constraints [
+    fields: [
+      type: {:custom, __MODULE__, :field_types, []},
```

Comment:
```elixir
  [
    fields: [
      type: :keyword_list,
      keys: [
        *: [
          type: :keyword_list,
          schema: [
            type: [
              type: Ash.OptionsHelpers.ash_type(),
              required: true
            ],
            constraints: [
              type: :keyword,
              default: []
            ]
          ]
        ]
      ]
    ]
  ]
```
Should be able to do something like this

Reply:
I tried this, but I'm a bit stuck because it does not compile with this error:

```
== Compilation error in file lib/ash/type/map.ex ==
** (FunctionClauseError) no function clause matching in NimbleOptions.Docs.get_raw_type_str/1    
    
    The following arguments were given to NimbleOptions.Docs.get_raw_type_str/1:
    
        # 1
        {:spark_type, Ash.Type, :builtins, [{"{:array, inner_type}", "list", "{:array, ${1:inner_type}}"}]}
    
    Attempted function clauses (showing 10 out of 27):
    
        defp get_raw_type_str(nil)
        defp get_raw_type_str({:custom, _mod, _fun, _args})
        defp get_raw_type_str(:mfa)
        defp get_raw_type_str(:mod_arg)
        defp get_raw_type_str({:or, _values})
        defp get_raw_type_str({:in, _})
        defp get_raw_type_str({:fun, arity})
        defp get_raw_type_str(:any)
        defp get_raw_type_str(:reference)
        defp get_raw_type_str(:pid)
        ...
        (17 clauses not shown)
    
    (nimble_options 1.0.1) lib/nimble_options/docs.ex:113: NimbleOptions.Docs.get_raw_type_str/1
    (nimble_options 1.0.1) lib/nimble_options/docs.ex:104: NimbleOptions.Docs.get_type_str/1
    (nimble_options 1.0.1) lib/nimble_options/docs.ex:43: NimbleOptions.Docs.option_doc/2
    (elixir 1.14.3) lib/enum.ex:2468: Enum."-reduce/3-lists^foldl/2-0-"/3
    (nimble_options 1.0.1) lib/nimble_options/docs.ex:63: NimbleOptions.Docs.option_doc/2
    (elixir 1.14.3) lib/enum.ex:2468: Enum."-reduce/3-lists^foldl/2-0-"/3
    (nimble_options 1.0.1) lib/nimble_options/docs.ex:6: NimbleOptions.Docs.generate/2
    lib/ash/type/map.ex:57: (module)
```

Reply:
This is the result of the Ash.OptionsHelpers.ash_type() call

Reply:
CI should release spark: v1.0.6 shortly.